src/objects.h

```diff
@@ -38,8 +39,10 @@
 **  'IS_INTOBJ' returns 1 if the object <o> is an (immediate) integer object,
 **  and 0 otherwise.
 */
-#define IS_INTOBJ(o) \
-    ((Int)(o) & 0x01)
+static inline Int IS_INTOBJ(Obj o)
+{
+    return ((Int)o & 0x03) == 0x01;
+}
```

Comment:
Also, I wonder if this has a performance impact (I'd assume it would be tiny and virtually not noticeable, but then again, a lot of code assumes that `IS_INTOBJ` is cheap as dirt...). Don't get me wrong, I am not opposed (to the contrary, I am all for this), just sayin' it for completness

So previously, this was a single `test` instruction on x86; now I assume it needs two instructions?